Us203 task214 : Connect Taiga/Taiga Tasks component to Taiga API for overview page

bpanda1 opened the pull request:
This is for overview page, I will do the same for each sprint in separate task so will cerate a new PR, once that is done.

Task status from Taiga is now showing the live data instead of hard-code value. Here these kind of tasks are shown:
- Completed
- Unassigned/New --> These tasks are created by scrum masters but no work has been started or not been assigned to anyone. This increases count in scrum master's data .
- In Progress
- Ready for Test

I tested the code and it is visible. I am not aware of any special comment method or something you guys are following, so please let me know those.

And also few console lines are left, I will remove those if I don't receive any comment about the change in current code.
<img width="552" alt="Screen Shot 2019-03-27 at 4 00 47 PM" src="https://user-images.githubusercontent.com/43048349/55117900-9de16400-50a9-11e9-9cbf-2f91bb48a457.png">

bpanda1 commented:
@HalcyonAura : I have changed one line in Taiga.ts file for Promise to work in the code previously written by me, so please feel free to put your comment about the change, if any.

trevorforrey commented on taigit/src/actions/taigaActions.js:
```diff
@@ -18,7 +19,10 @@ export const grabTaigaData = (infoForApiCall) => dispatch => {
    * 
    */
-  dispatch({type: GRAB_TAIGA_DATA, payload:'mockApiData'})
+  project_info('sanaydevi-ser-574') // give the slug name
```
I would change the hardcoded slug to a variable passed inplace of _infoForApiCall_ so it would be like:
```
export const grabTaigaData = (slugName) => dispatch => {
...
project_info(slugName)
...
```

bpanda1 replied:
Yeah. This is not the part of this task as well, I can revert this.

trevorforrey commented on taigit/src/actions/taigaActions.js:
```diff
@@ -28,3 +32,9 @@ export const grabSprintStats = (infoForApiCall) => dispatch => {
       dispatch({type: GET_SPRINT_STATS, payload: sprintStats});
     });
 }
+
```
Same comment as above. Moving hardcoded value to passed variable.

bpanda1 replied:
This is not part of this task, I will revert the changes.. I was doing code walk through and did changes to understand the flow. :P

trevorforrey commented on taigit/src/reducers/taigaReducer.js:
```diff
@@ -59,18 +66,73 @@ export const selectSprintProgressChartData = (state) => {
+  let taigaTaskStatsArr = state.taigaTaskStats;
+  let dictToProcessData = {};
+  // Deserialzation of received data from backend
+  for (let i = 0; i < taigaTaskStatsArr.length; i++) {
```
It looks like this grabs data for the entirety of the project, this is great, and we can use this in the _Overview_ page. For the Taiga page, we're going to have the data display based on a sprint selected in the dropdown menu. Would you be able to add another selector function, that takes in a name or id of a sprint and then returns chart data for only that sprint?

bpanda1 replied:
Sure, I can do that. Shall I create a new task to implement this with specific sprint or should just go with this one?

trevorforrey commented:
One last thing! I would remove the _unassigned tasks_ as it only shows who created the task.

bpanda1 commented on taigit/src/reducers/taigaReducer.js:
```diff
@@ -59,18 +66,73 @@ export const selectSprintProgressChartData = (state) => {
 }
 
 export const selectUserTaskDistributionChartData = (state) => {
+  var backgroundColor = [
```
okay! I will look in reducers/githubReducer.js file.

bpanda1 replied:
Done! Changed it as per color.js. And also added color green to show completed Task.

bpanda1 commented:
I added the unassigned Tasks separately in the X- axis as discussed. 
<img width="549" alt="Screen Shot 2019-03-29 at 9 54 49 AM" src="https://user-images.githubusercontent.com/43048349/55249507-7a352f80-5209-11e9-8a1a-ff6974ced3a1.png">

bpanda1 commented:
Trevor, Moved this to overview page and removed it from Taiga page, so that sprint Task distribution can be shown there.
<img width="553" alt="Screen Shot 2019-03-30 at 5 45 32 PM" src="https://user-images.githubusercontent.com/43048349/55283293-bef9bd00-5314-11e9-99a1-31c2e3becaf3.png">

trevorforrey commented on taigit/src/actions/taigaActions.js:
```diff
@@ -28,3 +32,9 @@ export const grabSprintStats = (infoForApiCall) => dispatch => {
 }
+
+export const grabTaskStats = (infoForApiCall) => dispatch => {
+  get_task_status_count(306316).then((taskStats) => {
```
I would have this as a parameter that's passed into _grabTaskStats_ 👍

trevorforrey commented on taigit/src/components/Overview.js:
```diff
@@ -4,12 +4,17 @@ import {Doughnut, Line} from 'react-chartjs-2';
+import { Bar } from 'react-chartjs-2';
+import { selectUserTaskDistributionChartData } from '../reducers';
+import { connect } from 'react-redux';
+import { grabTaigaData, grabSprintStats, grabTaskStats } from '../actions/taigaActions';
```
I believe you can remove the import of _grabTaigaData_ and _grabSprintStats_, since we only need to call _grabTaskStats_ here

bpanda1 commented:
Trevor, I think I have taken care of all your comments and resolved the conflict as well. Let me know if I missed anything. :)

trevorforrey commented:
Looks good! Feel free to merge in whenever!
